app/src/main/java/com/guru/recognizetext/utils/ImageFileManager.kt

```diff
@@ -1,30 +1,30 @@
-package com.guru.recognizetext.helperclasses
+package com.guru.recognizetext.utils
 
 import android.content.Context
 import android.net.Uri
 import android.os.Environment
 import androidx.core.content.FileProvider
 import java.io.File
+import javax.inject.Inject
 
-class ImageFileManager(private val context: Context) {
+class ImageFileManager @Inject constructor(private val context: Context ) {
     private lateinit var tempFile: File
-    lateinit var imageUri: Uri
-
-    fun generateImageUri() {
-        imageUri = FileProvider.getUriForFile(
-            context,
-            context.packageName + ".fileprovider",
-            tempFile
-        )
-    }
+     lateinit var imageUri: Uri
 
     fun createTempFile() {
         tempFile = File.createTempFile(
             "image",
```

Comment:
Are these files being permanently saved in the user's directory? If so, then we might want to choose unique names for all these files by appending `image` with some unique value. A reasonably safe choice could be to prepend `image` with the current system time milliseconds.

Reply:
is it right way to create unique names for this image
fun createTempFile() {
    val uniqueValue = System.currentTimeMillis().toString()
    tempFile = File.createTempFile(
        "image_$uniqueValue",
        ".jpg",
        getExternalFilesDir(Environment.DIRECTORY_PICTURES)
    )
}

Reply:
Yes.. this looks good to me 👍